src/realm/deppart/byfield_gpu_impl.hpp

```diff
@@ -0,0 +1,261 @@
+        /* num_items */       num_valid_rects, stream);
+
+
+    size_t total_pts;
+    CUDA_CHECK(cudaMemcpyAsync(&total_pts, &d_prefix_rects[num_valid_rects], sizeof(size_t), cudaMemcpyDeviceToHost, stream), stream);
+
+    CUDA_CHECK(cudaStreamSynchronize(stream), stream);
+
+    nvtx_range_pop();
+
+    nvtx_range_push("cuda", "build points");
+
+    RegionInstance points_instance = this->realm_malloc(total_pts * sizeof(PointDesc<N,T>), my_mem);
+    PointDesc<N,T>* d_points = reinterpret_cast<PointDesc<N,T>*>(AffineAccessor<char,1>(points_instance, 0).base);
+
+    FT* d_colors;
+    RegionInstance colors_instance;
+
+    if constexpr(std::is_same_v<FT,bool>) {
+      std::vector<uint8_t> flat_colors(colors.size());
+      for (size_t i = 0; i < colors.size(); i++) {
+        flat_colors[i] = colors[i] ? 1 : 0;
+      }
+      colors_instance = this->realm_malloc(total_pts * sizeof(PointDesc<N,T>), my_mem);
+      uint8_t* d_flat_colors = reinterpret_cast<uint8_t*>(AffineAccessor<char,1>(colors_instance, 0).base);
+      CUDA_CHECK(cudaMemcpyAsync(d_flat_colors, flat_colors.data(), colors.size() * sizeof(uint8_t), cudaMemcpyHostToDevice, stream), stream);
+      d_colors = reinterpret_cast<FT*>(d_flat_colors);
+    } else {
+      colors_instance = this->realm_malloc(colors.size() * sizeof(FT), my_mem);
+      d_colors = reinterpret_cast<FT*>(AffineAccessor<char,1>(colors_instance, 0).base);
+      CUDA_CHECK(cudaMemcpyAsync(d_colors, colors.data(), colors.size() * sizeof(FT), cudaMemcpyHostToDevice, stream), stream);
+    }
+
+    AffineAccessor<FT,N,T> h_accessors[field_data.size()];
+    for (size_t i = 0; i < field_data.size(); ++i) {
+      h_accessors[i] = AffineAccessor<FT,N,T>(field_data[i].inst, field_data[i].field_offset);
+    }
+
+    RegionInstance accessors_instance = this->realm_malloc(field_data.size() * sizeof(AffineAccessor<FT,N,T>), my_mem);
+    AffineAccessor<FT,N,T>* d_accessors = reinterpret_cast<AffineAccessor<FT,N,T>*>(AffineAccessor<char,1>(accessors_instance, 0).base);
+    CUDA_CHECK(cudaMemcpyAsync(d_accessors, h_accessors, field_data.size() * sizeof(AffineAccessor<FT,N,T>), cudaMemcpyHostToDevice, stream), stream);
+
+    nvtx_range_pop();
+
+    nvtx_range_push("cuda", "populate bitmasks");
+
+    int threads_per_block = 256;
+    int num_blocks = (total_pts + threads_per_block - 1) / threads_per_block;
+    byfield_gpuPopulateBitmasksKernel<N,T,FT><<<num_blocks, threads_per_block, 0, stream>>>(d_accessors, d_valid_rects, d_prefix_rects, d_inst_prefix, d_colors, total_pts, colors.size(), num_valid_rects, field_data.size(), d_points);
+    CUDA_CHECK(cudaStreamSynchronize(stream), stream);
+    KERNEL_CHECK(stream);
+
+    nvtx_range_pop();
+
+  std::map<FT, size_t> color_indices;
+  for (size_t i = 0; i < colors.size(); i++) {
+    color_indices[colors[i]] = i;
+  }
+
+  CUDA_CHECK(cudaStreamSynchronize(stream), stream);
+  cudaStreamDestroy(stream);
+  inst_entries_instance.destroy();
```

Comment:
There are instances you deleted here but not in the exit of the fast path. Also, do you need to wait until the end of the function to delete these? You should be trying to free (or reuse) these instances as soon as the data is no longer needed.